hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceV1.java

```diff
@@ -110,6 +111,30 @@
         registerJavaTypeSerializers();
     }
 
+    @Override
+    public <B extends Data> B toData(Object obj, DataType type) {
```

Comment:
i remember @jerrinot had some objections to add a this kind of methods but not sure if he keeps them still. We opened a related issue previously: https://github.com/hazelcast/hazelcast/issues/10249